### ISSUE TITLE
Add an option to exclude classes by name pattern in LargeClass rule

### ISSUE THREAD

**Issue by magiepooh** [detekt/detekt#8911]
## Expected Behavior of the rule

I am suggesting an enhancement to the existing LargeClass rule within the Complexity rule set.

The current behavior of LargeClass is to flag any class exceeding the configured maximumLineCount. The desired change is to introduce a configuration setting in detekt.yml that allows users to exclude specific classes or class name patterns from this check.

Proposed New Configuration:
The rule should respect an exclusion pattern defined in detekt.yml.

```yaml
complexity:
  LargeClass:
    active: true
    maximumLineCount: 600
    # Proposed new option to exclude classes matching a name pattern
    excludeClassPattern: '.*Module|.*Config|.*Constants|.*Test|.*Util|.*Helper'
```

## Context

We are seeking a more practical and maintainable quality gate for class size.

Currently, warnings must be suppressed using @Suppress("LargeClass") directly in the source code for classes where high line counts are conventionally accepted or architecturally necessary.

Key examples where exclusion is needed include:
- Test Classes (*Test): They often contain numerous small, focused test functions, resulting in high line counts even if each test adheres to SRP.
- Utility/Helper Classes (*Util, *Helper): Used to consolidate many related extension functions or helper methods.
- Configuration Classes (*Module, *Config): They grow large due to necessary dependency injection bindings or application settings.

Introducing an excludeClassPattern option will allow for centralized configuration in detekt.yml, keeping the source code clean and aligning this rule with other detekt rules that already support class exclusion patterns.

**Comment by magiepooh** [comment]
Still relevant. I'm still encountering this issue (or interested in this feature).

**Comment by BraisGabin** [comment]
You can do that with `exclude`. You wouldn't exclude classes but files but I guess that if you have a big class you don't have more classes on that file
